velox/dwio/common/SelectiveStructColumnReader.cpp

```diff
@@ -99,7 +99,7 @@ void setConstantField(
     const VectorPtr& constant,
     vector_size_t size,
     VectorPtr& field) {
-  if (field && field->isConstantEncoding() && field.unique() &&
+  if (field && field->isConstantEncoding() && field.use_count() == 0 &&
```

Comment:
We don't seem to use () around boolean expressions like these within if statements. So I didn't add it. I just added it when using macros where the argument is expanded to avoid any potential issues.